### PR TITLE
Don't convert bytes headers to str

### PR DESCRIPTION
**Pull request by h3llrais3r**
Backport https://github.com/cherrypy/cherrypy/pull/1736
Also fixes https://github.com/Lawouach/WebSocket-for-Python/pull/252

**What kind of change does this PR introduce?**
  - [x] bug fix

### CODE REVIEW

**Comment by webknjaz** [comment]
Hey @h3llrais3r could you please fix tests?

**Comment by h3llrais3r** [comment]
@webknjaz Can you check again?
I've fixed the travis build (flake8 error) but still errors for pydocstyle (not related to my changes) and tests against latest Python nightly.
As far as I can see, the AppVeyor tests are working fine for Python 2.7, Python 3.5 and Python 3.6. Python 3.5 is not working, but that's not related to my changes...

File: cherrypy/lib/httputil.py [hunk 1]
```diff
@@ -518,7 +518,8 @@ def encode_header_items(cls, header_items):
         transmitting on the wire for HTTP.
         """
         for k, v in header_items:
-            if not isinstance(v, six.string_types):
+            if not isinstance(v, six.string_types) and \
```
Comment (webknjaz) [inline cherrypy/lib/httputil.py:521]: Please use brackets instead. Escaping EOL looks ugly and we don't use it.

Comment (webknjaz) [reply cherrypy/lib/httputil.py:521]: And this probably only needs `binary_type` check.

File: cherrypy/test/test_encoding.py [hunk 2]
```diff
@@ -50,6 +50,8 @@ def cookies_and_headers(self):
                 cherrypy.response.cookie['candy']['domain'] = 'cherrypy.org'
                 cherrypy.response.headers[
                     'Some-Header'] = 'My d\xc3\xb6g has fleas'
```
Comment (webknjaz) [inline cherrypy/test/test_encoding.py:52]: Maybe add u-prefix here?

File: cherrypy/lib/httputil.py (hunk 1 above)
Comment (h3llrais3r) [reply cherrypy/lib/httputil.py:521]: It is actually used in the code. See https://github.com/cherrypy/cherrypy/blob/maint/17.x/cherrypy/_cpdispatch.py#L345-L347.
I based myself on existing examples, that's why I did it like this.
Where do you want brackets exactly?
I only added the check for `binary_type`, I did not touch the existing check for `string_types`.

File: cherrypy/test/test_encoding.py (hunk 2 above)
Comment (h3llrais3r) [reply cherrypy/test/test_encoding.py:52]: No u-prefix is used at other places in this test.
This is also not something I've added. I only added the byte header.
So do you still want me to add it or can we leave it like this?

File: cherrypy/lib/httputil.py (hunk 1 above)
Comment (h3llrais3r) [reply cherrypy/lib/httputil.py:521]: So basically, I applied the same fix (and same test) as in the original fix.

File: cherrypy/test/test_encoding.py (hunk 2 above)
Comment (jaraco) [reply cherrypy/test/test_encoding.py:52]: Right - definitely doesn't make sense to be making editorial changes in unaffected code.

**Pull request merged**